ambry-commons/src/test/java/com.github.ambry.commons/CopyingAsyncWritableChannelTest.java

```diff
@@ -56,6 +73,41 @@ public void basicsTest() throws Exception {
     writeAndCheckCallback(ByteBuffer.allocate(0), channel, 0, ClosedChannelException.class, null);
   }
 
+  /**
+   * Test that {@link CopyingAsyncWritableChannel} behaves as expected: chunks are copied, callback completed
+   * immediately after {@link CopyingAsyncWritableChannel#write} method completes.
+   */
+  @Test
+  public void basicsTestWithNettyByteBuf() throws Exception {
+    for (boolean useCompositeByteBuf : Arrays.asList(false, true)) {
```

Comment:
Is this meant to test the default method in `AsyncWritableChannel`? We could also consider overriding the write(ByteBuf) method for `CopyingAsyncWritableChannel`, which should be relatively straightforward since it just copies data to byte arrays internally.

Reply:
you are right. The implementations are driven by the need. There is no need for CopyingAsyncWritableChannel to implement write(ByteBuf) for now. But later we will have to change all the implementations for it.